assets/js/blocks/featured-items/image-editor.tsx

```diff
@@ -45,24 +46,70 @@ type WithImageEditorProps< T extends EditorBlock< T > > =
 	| ( T & WithImageEditorProductProps< T > );
 
 interface ImageEditorProps {
+	align: string;
 	backgroundImageId: number;
 	backgroundImageSize: MediaSize;
 	backgroundImageSrc: string;
+	containerRef: React.RefObject< HTMLDivElement >;
 	isEditingImage: boolean;
 	setAttributes: ( attrs: MediaAttributes ) => void;
 	setIsEditingImage: ( value: boolean ) => void;
 }
 
+// Adapted from:
+// https://github.com/WordPress/gutenberg/blob/v15.6.1/packages/block-library/src/image/use-client-width.js
+function useClientWidth(
+	ref: React.RefObject< HTMLDivElement >,
+	dependencies: string[]
+) {
+	const [ clientWidth, setClientWidth ]: [
+		number | undefined,
+		Dispatch< SetStateAction< number | undefined > >
+	] = useState();
+
+	const calculateClientWidth = useCallback( () => {
+		setClientWidth( ref.current?.clientWidth );
+	}, [ ref ] );
+
+	useEffect( calculateClientWidth, [
+		calculateClientWidth,
+		...dependencies,
+	] );
+	useEffect( () => {
+		if ( ! ref.current ) {
+			return;
+		}
+		const { defaultView } = ref.current.ownerDocument;
+
+		if ( ! defaultView ) {
+			return;
+		}
+		defaultView.addEventListener( 'resize', calculateClientWidth );
+
+		return () => {
+			defaultView.removeEventListener( 'resize', calculateClientWidth );
+		};
+	}, [ ref, calculateClientWidth ] );
+
+	return clientWidth;
+}
+
 export const ImageEditor = ( {
+	align,
 	backgroundImageId,
 	backgroundImageSize,
 	backgroundImageSrc,
+	containerRef,
 	isEditingImage,
 	setAttributes,
 	setIsEditingImage,
 }: ImageEditorProps ) => {
-	return (
-		<>
+	const clientWidth = useClientWidth( containerRef, [ align ] );
```

Comment:
This function could be moved below the fallback for WP 6.1. In that case we don't new to attach the event listeners and make all the computations that happen in scope of `useClientWidth`. What do you think?

Reply:
I don't think we can because it's a React hook so it can't be called conditionally:

https://legacy.reactjs.org/docs/hooks-rules.html

Reply:
Oh, of course. Missed that! 👍